packages/utils/src/collections.ts

```diff
@@ -7,3 +7,13 @@ export function zip<A, B, C>(fn: (a: A, b: B) => C, as: A[], bs: B[]) {
   }
   return res
 }
+
+export function zip3<A, B, C>(as: A[], bs: B[], cs: C[]) {
```

Comment:
If you are interested, you can do a more general implementation using [function overloads](https://www.typescriptlang.org/docs/handbook/functions.html#overloads). More general in the sense that we can make it work for 2 or more parameters, and still return an `Array<tuple>`.

Probably we should rename the other zip to `zipWith`

Reply:
Yeah I was curious, but didn't want to spend too much time on this :P

Reply:
nevermind then ;)